arelle/XmlUtil.py

```diff
@@ -875,15 +1125,22 @@ def xmlstring(elt, stripXmlns=False, prettyPrint=False, contentsOnly=False, incl
         return _text + ('\n' if prettyPrint else '').join(
             xmlstring(child, stripXmlns, prettyPrint)
             for child in elt.iterchildren()) + _tail
-    xml = etree.tostring(elt, encoding=_STR_UNICODE, pretty_print=prettyPrint)
+    xml: str = etree.tostring(elt, encoding=_STR_UNICODE, pretty_print=prettyPrint)  # type: ignore[name-defined] # builtins
     if not prettyPrint:
         xml = xml.strip()
     if stripXmlns:
         return xmlnsStripPattern.sub('', xml)
     else:
         return xml
 
-def writexml(writer, node, encoding=None, indent='', xmlcharrefreplace=False, parentNsmap=None):
+def writexml(
+    writer: Any,
```

Comment:
I think this can be typed as `typing.TextIO`:
```suggestion
    writer: TextIO,
```